Migrate plugin host/set/catalog prefixes to typed

jefferai opened the pull request.

johanbrandhorst commented on internal/host/plugin/ids.go:
```diff
@@ -10,16 +10,19 @@ import (
-	HostCatalogPrefix = "hc"
-	HostSetPrefix     = "hs"
-	HostPrefix        = "h"
+	HostCatalogPrefix         = "hcplg"
+	PreviousHostCatalogPrefix = "hc"
+	HostSetPrefix             = "hs"
+	PreviousHostSetPrefix     = "hsplg"
+	HostPrefix                = "h"
+	PreviousHostPrefix        = "hplg"
```
The choice of naming here seems inconsistent, do we really want the _new_ `HostCatalog` prefix to include the `plg` suffix when the other new ones don't?

jefferai replied:
Oh weird

How did that happen

johanbrandhorst commented on internal/daemon/controller/handlers/targets/target_service.go:
```diff
@@ -1665,7 +1665,7 @@ func validateAddSetsRequest(req *pbs.AddTargetHostSetsRequest) error {
 	}
 	for _, id := range req.GetHostSetIds() {
 		if !handlers.ValidId(handlers.Id(id), static.HostSetPrefix) &&
-			!strings.HasPrefix(id, fmt.Sprintf("%s_", plugin.HostSetPrefix)) {
+			!(strings.HasPrefix(id, fmt.Sprintf("%s_", plugin.HostSetPrefix)) || strings.HasPrefix(id, fmt.Sprintf("%s_", plugin.PreviousHostSetPrefix))) {
```
Minor improvement maybe?
```go
prefix, _, _ := strings.Cut(id, "_")
if !handlers.ValidId(handlers.Id(id), static.HostSetPrefix) && prefix != plugin.HostSetPrefix && prefix != plugin.PreviousHostSetPrefix

louisruch commented on internal/credential/static/public_ids.go:
```diff
@@ -10,18 +10,20 @@ import (
-	CredentialPrefix      = "cred"
+	CredentialStorePrefix         = "csst"
+	PreviousCredentialStorePrefix = "cs"
+	CredentialPrefix              = "credst"
```
This credential is actually referring to the `username_password` credential, each type should get it's own prefix as they have different attributes. So something like `cup` or `credup`

jefferai commented:
@louisruch @johanbrandhorst After discussion with @louisruch we realized some more changes needed to be made. I've also simplified the logic -- I had added to logic that was there in target_service.go that already was doing the wrong thing -- thanks for the pointer on that @johanbrandhorst .

louisruch reviewed (approved):
LGTM
